File: exercise/day05/src/main/java/people/Person.java
package people;

import java.util.ArrayList;
import java.util.List;

public record Person(String firstName, String lastName, List<Pet> pets) {
    public Person(String firstName, String lastName) {
        this(firstName, lastName, new ArrayList<>());
    }

    public Person addPet(PetType petType, String name, int age) {
        pets.add(new Pet(petType, name, age));
        return this;
    }
}

File: solution/c#/Day03/Day03/Person.cs
namespace Day03
{
    public record Person(string FirstName, string LastName, params Pet[] Pets);

    public record Pet(PetType Type, string Name, int Age);

    public enum PetType
    {
        Cat,
        Dog,
        Hamster,
        Turtle,
        Bird,
        Snake
    }
}